src/dynarmic/common/x64_disassemble.cpp

```diff
@@ -21,15 +21,36 @@ void DumpDisassembledX64(const void* ptr, size_t size) {
 
     size_t offset = 0;
     ZydisDecodedInstruction instruction;
-    while (ZYAN_SUCCESS(ZydisDecoderDecodeBuffer(&decoder, (const char*)ptr + offset, size - offset, &instruction))) {
+    while (ZYAN_SUCCESS(ZydisDecoderDecodeBuffer(&decoder, static_cast<const char*>(ptr) + offset, size - offset, &instruction))) {
         fmt::print("{:016x}  ", (u64)ptr + offset);
 
         char buffer[256];
-        ZydisFormatterFormatInstruction(&formatter, &instruction, buffer, sizeof(buffer), (u64)ptr + offset);
+        ZydisFormatterFormatInstruction(&formatter, &instruction, buffer, sizeof(buffer), reinterpret_cast<u64>(ptr) + offset);
         puts(buffer);
 
         offset += instruction.length;
     }
 }
 
+std::vector<std::string> Dynarmic::Common::DisassembleX64(const void* ptr, size_t size) {
```

Comment:
```suggestion
std::vector<std::string> DisassembleX64(const void* ptr, size_t size) {
```

This should allow the builders to succeed